src/test/java/com/google/sps/data/JourneyTest.java

```diff
@@ -0,0 +1,333 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.sps.data;
+
+import com.google.maps.errors.ApiException;
+import java.lang.Math;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.io.IOException;
+import java.lang.InterruptedException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests the behaviour of a Journey object.
+ * TODO[ak47na]: Implement brute-force algorithm that iterates over all permutations of waypoints
+ * and finds the optimal one.
+ */
+@RunWith(JUnit4.class)
+public class JourneyTest {
+  private ManhattanDistancePathFinder pathFinder = new ManhattanDistancePathFinder();
+  @Test
+  public void testOptimalTimeJourney1() throws ApiException, BadRequestException, DataNotFoundException, IOException, InterruptedException {
```

Comment:
Try to work out in the test name how does testOptimalTimeJourney1 differ from testOptimalTimeJourney1. In general, each test should be testing a single specific behavior, and you should feel somewhat confident that the code works correctly if it produces the right result.

Reply:
I changed the names to `testOptimalTimeJourneyRestrictionsDoNotAffectOptimalOrder` and `testOptimalTimeJourneyRestrictionsAffectOptimalOrder`